### DISCUSSION TITLE
on_command装饰器无法生效

### DISCUSSION THREAD

**Discussion by yzblacia** [nonebot/discussions#148] [问答]
![@}A`%9J9PWC0CNCK`HX}M7J](https://user-images.githubusercontent.com/93920506/169080472-56c34ba1-c893-4e1d-a3de-8c2c2241b546.png)
![$~WU6 {O(2 ITGYG~N_Z6SK](https://user-images.githubusercontent.com/93920506/169080719-ee89fe27-3666-4dfc-b7df-0817dfa49e02.png)
![$EQATP_H _ HK{LW78}15GF](https://user-images.githubusercontent.com/93920506/169080750-aa4430ed-16b8-4929-9f5b-037d84aa3131.png)
![XH}9O8_1NE(_ }X5WZVZ) Q](https://user-images.githubusercontent.com/93920506/169080787-ccd7fa93-6f44-46a2-aa4e-022bbcb0ccc5.png)
我尝试了nonebot官方文档的实例函数，第一次使用时on_command函数正确添加了我想要的命令，然后我修改了plugin里的weather.py里的函数，重新启动bot，但是bot返回给我的命令依然是一开始的，也就是说weather.py里我修改的函数根本没有生效，on_command没有重新加载我重新定义的命令和别名，这是为什么呢

**Comment by yanyongyu** [comment]
你是不是重复写了两个，看看command.py和weather.py

**Reply by yzblacia** [reply]
谢谢你的答复。并没有，我command.py是空文件，应该不是这个原因。如果你愿意帮我解决问题的话，我可以把代码文件发给你看看，谢谢！

**Reply by yanyongyu** [reply]
看你的日志中warning就是你有两个相同命令

**Reply by yzblacia** [reply]
![@IOL$LR1ADM61H)`3O)J)S8](https://user-images.githubusercontent.com/93920506/169093852-cd77bd49-7892-4b63-89fd-e25cf31f9844.png)
![V$TP739IM79O70I{VMOO9UT](https://user-images.githubusercontent.com/93920506/169094013-6521954a-5a5c-46dd-ac03-5ec29572f0ef.png)
我刚刚又去重新跑了几遍代码，现在是没有报错了，代码正常运行。
啥都没改，不明白一开始为啥出问题。
谢谢你的答复。
